tinyiiod.c

```diff
@@ -62,160 +74,219 @@ int tinyiiod_read_command(struct tinyiiod *iiod)
 	if (bytes > sizeof(buf))
 		bytes = sizeof(buf);
-
 	tinyiiod_read(iiod, buf, bytes);
-	buf[bytes] = '\0';
+
+	buf[bytes - 1] = '\0';
 
 	if (channel)
 		ret = iiod->ops->ch_write_attr(device, channel, ch_out,
-				attr, buf, bytes);
+					       attr, buf, bytes);
 	else
 		ret = iiod->ops->write_attr(device, attr, buf, bytes, debug);
 
 	tinyiiod_write_value(iiod, (int) ret);
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_do_open
+*******************************************************************************/
 void tinyiiod_do_open(struct tinyiiod *iiod, const char *device,
-		size_t sample_size, uint32_t mask)
+		      size_t sample_size, uint32_t mask)
 {
-	int ret = iiod->ops->open(device, sample_size, mask);
+	int32_t ret = iiod->ops->open(device, sample_size, mask);
 	tinyiiod_write_value(iiod, ret);
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_do_close
+*******************************************************************************/
 void tinyiiod_do_close(struct tinyiiod *iiod, const char *device)
 {
-	int ret = iiod->ops->close(device);
+	int32_t ret = iiod->ops->close(device);
 	tinyiiod_write_value(iiod, ret);
 }
 
+/***************************************************************************//**
+ * @brief tinyiiod_do_exit
+*******************************************************************************/
+int32_t tinyiiod_do_exit(struct tinyiiod *iiod)
+{
+	return iiod->ops->exit(iiod->instance_id);
+}
+
+/***************************************************************************//**
+ * @brief tinyiiod_do_writebuf
+*******************************************************************************/
+int32_t tinyiiod_do_writebuf(struct tinyiiod *iiod,
+			     const char *device, size_t bytes_count)
+{
+	char *pbuffer = (char*)malloc(bytes_count);
+	if(pbuffer == NULL) {
+		return -ENOMEM;
+	}
+	memset(pbuffer, 0, bytes_count);
+	tinyiiod_read_nonblocking(iiod, pbuffer, bytes_count);
+	tinyiiod_write_value(iiod, (int) bytes_count);
+	size_t bytes_received = tinyiiod_read_wait(iiod, bytes_count);
+	if(bytes_count != bytes_received) {
+		return -EPIPE;
+	}
+
+	iiod->ops->write_device(device, pbuffer, bytes_count);
+	tinyiiod_write_value(iiod, (int) bytes_count);
+	free(pbuffer);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief tinyiiod_do_readbuf
+*******************************************************************************/
 void tinyiiod_do_readbuf(struct tinyiiod *iiod,
-		const char *device, size_t bytes_count)
+			 const char *device, size_t bytes_count)
 {
-	int ret;
-	char buf[256];
+	int32_t ret;
+	char *pbuf; // pointer to memory, where DMA writes, memory free not necessary
 	uint32_t mask;
-	bool print_mask = true;
 
 	ret = iiod->ops->get_mask(device, &mask);
 	if (ret < 0) {
 		tinyiiod_write_value(iiod, ret);
 		return;
 	}
+	// read device
```

Comment:
comments should all be consistent with C style like /* Something */.

Reply:
@amiclaus 
can astyle handle this ? i.e. complaining about C style comments ?
i do feel this is something we could automate